File: 0119.杨辉三角II/0119-杨辉三角II.rs
impl Solution {
    pub fn get_row(n: i32) -> Vec<i32> {
        let mut ret: Vec<Vec<i32>> = Vec::new();
        for i in 1..((n+2) as usize) {
            let mut arr = vec![1;i];
            for j in 1..i-1 {
                arr[j] = ret[i-2][j-1] + ret[i-2][j];
            }
            ret.push(arr);
        }
        return ret[n as usize].clone();
        
    }
}

File: 0119.杨辉三角II/0119-杨辉三角II.py
class Solution:
    def getRow(self, n: int) -> List[int]:
        if n == 0:
            return [1]
        if n == 1:
            return [1, 1]
        pre = [1, 1]
        ret = []
        for _ in range(n - 1):
            len_ = len(pre) + 1
            ret = [1] * len_
            for i in range(1, len(ret) - 1):
                ret[i] = pre[i] + pre[i - 1]
            pre = ret[:]
        return ret
